web/src/components/DeleteSemesterAlertDialog.tsx

```diff
@@ -0,0 +1,68 @@
+import {
+  AlertDialog,
+  AlertDialogBody,
+  AlertDialogContent,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogOverlay,
+  Button,
+  useToast
+} from '@chakra-ui/react'
+import React from 'react'
+import { useDeleteSemesterMutation } from '../generated/graphql'
+import { toastApolloError } from '../util'
+
+interface Props {
+  isOpen: boolean,
+  onClose: () => void,
+  semesterId: string,
+  name: string
+}
+
+export const DeleteSemesterAlertDialog: React.FC<Props> = ({ isOpen, onClose, semesterId, name }) => {
+  const toast = useToast()
+
+  const [remove] = useDeleteSemesterMutation({
+    onError: errors => toastApolloError(toast, errors)
+  })
+
+  return (
+    <AlertDialog isOpen={isOpen} onClose={onClose} leastDestructiveRef={undefined} isCentered>
```

Comment:
Was hat es mit leastDestructiveRef auf sich?

Reply:
ist anscheind ein required attribut, welches das Element bestimmt welches nach dem Schießen des Alerts gefocused wird. Gibt es auch für das normale Modal als `finalFocusRef`, see https://chakra-ui.com/docs/overlay/modal#keyboard-and-focus-management